app/models/member.rb

```diff
@@ -4,22 +4,23 @@
 #
 # Table name: members
 #
-#  id                :integer          not null, primary key
-#  email             :string
-#  country           :string
-#  first_name        :string
-#  last_name         :string
-#  city              :string
-#  postal            :string
-#  title             :string
-#  address1          :string
-#  address2          :string
-#  created_at        :datetime         not null
-#  updated_at        :datetime         not null
-#  actionkit_user_id :string
-#  donor_status      :integer          default("0"), not null
-#  more              :jsonb
-#  consented_at      :datetime
+#  id                   :integer          not null, primary key
+#  email                :string
+#  country              :string
+#  first_name           :string
+#  last_name            :string
+#  city                 :string
+#  postal               :string
+#  title                :string
+#  address1             :string
+#  address2             :string
+#  created_at           :datetime         not null
+#  updated_at           :datetime         not null
+#  actionkit_user_id    :string
+#  donor_status         :integer          default("0"), not null
+#  more                 :jsonb
+#  consented_updated_at :datetime
```

Comment:
What's the logic in making these two fields instead of `consented_at` - is the idea that if someone revokes consent, `consented` will be `false` and `consented_updated_at` will be the timestamp of the update? 

I would like for us to be a bit more mindful with the naming. If we could be explicit about `consented` meaning GDPR consent and name it something like `gdpr_consented`. 

If we want to keep both the status of the consent and timestamps of its changes, it would make sense for GDPR consents to be in a table of their own with `created_at` `updated_at`, and an enum `status` with `active` or `revoked`. If we ever need to make GDPR consent more granular (which is honestly what is required of us - members should get to choose what we're allowed to do and mailing list consent would be different from cookie consent), being able to add new logic to a separate table would make things easier.

Reply:
I'm keen on the idea of a separate table where we can keep a record of every change, which would allow us to attach metadata about when it happened (e.g. what action triggered it).

Reply:
**On having two fields:** The logic behind having `consented_updated_at` and `consented`, is mainly being able to record refusals to consent. Having only a timestamp doesn't make it possible to differentiate not giving consent (from never getting the form) from explicitly clicking "no, I don't consent". 

**On naming:** I agree that `consented` it not good enough a name,  but I also don't think we should use `gdpr` as part of the naming, since it's too broad a term. GDPR regulates a lot of stuff, and what we mean here is "consent to send emails". What do you think about renaming `consented` to `mailing_consent`?

**On havin a separate table** I agree that having a separate table (or an array field in the members table) to track changes in consent would be ideal, but I think this solution is good enough for now, and I'd like to keep the scope to a minimum to make sure we make it to the 25th deadline.

Reply:
Let's remember not to lose all of the members with a `consented_at` value when this goes to production.